changelog/no-cmpsb.dd

```diff
@@ -0,0 +1,24 @@
+# `memcmp()` compares are no longer performed for struct equality tests, memberwise comparisons are done instead, per the spec
+
+The compiler would sometimes generate code for struct equality tests using
+`memcmp()` across the whole struct object. This assumed that:
+
+1. alignment holes were filled with 0, which is not always the case
+
+2. there were no floating point NaN values, which should compare as not equal
+even if the bit patterns match
+
+The spec requires that the comparison be done in a memberwise manner.
+This brings the implementation in line with this.
+
+https://dlang.org/spec/expression.html#equality_expressions
```

Comment:
FWIW this link won't be clickable because Ddoc doesn't support this. You need to use LINK2 or similar macros.

Reply:
Ddoc should be auto-recognizing URLs.